common/src/main/java/com/frostwire/util/Ref.java

```diff
@@ -49,6 +49,10 @@ private Ref() {
     }
 
     public static void free(Reference<?> ref) {
-        ref.clear();
+        try {
+            ref.clear();
+        } catch (Throwable t) {
```

Comment:
interesting...did you get an exception here? (Regards!)

Reply:
I believe I did

Reply:
Alden, le eché una limpiada grandisima al MusicPlaybackService y todo Apollo.

Nada de tener un timer para decidir si apagar el servicio, ahora es todo lo contrario.

El servicio está apagado a menos que quieras tocar musica, y se acabo el guevito de tener "FrostWire is Running" todo el tiempo, y acabando con "not called startedForeground" ANRs

Con respecto al servicio de BitTorrent, ahora es un JobIntentService, y funciona bien, Android no es tan agresivo matando el servicio, y no está corriendo todo el tiempo.

Reply:
Esta bueno eso de usar el `JobIntentService`, a ver si un día de estos puedo hacer algún experimento.